### PR TITLE
Ratatouille example

### PR DESCRIPTION
**Pull request by mhanberg**
This example demonstrates how to use the [Ratatouille](https://github.com/ndreynolds/ratatouille) library to build a non-trivial TUI.

This specific example ports a basic version of the popular [fzf](https://github.com/junegunn/fzf) tool, specifically a half implementation of its V1 algorithm.

https://user-images.githubusercontent.com/5523984/177672707-4175d660-6372-4f19-82bd-22cf1e020e19.mov

### CODE REVIEW

**Comment by wojtekmach** [comment]
This is AMAZING. I’m a little bit hesitant to merge this in though because I kind of want to keep these example scripts as minimal as possible. I’d even take some of these, make minor adjustments, and use in bug reports so that it is trivial for the maintainer to reproduce. So I think I’d be more keen on something minimal for this library too. But at the same time this example script is so incredibly cool…

**Comment by mhanberg** [comment]
@wojtekmach no worries! I have a version in my dotfiles (https://github.com/mhanberg/.dotfiles/blob/main/bin/fxf), so the example will live on regardless.

A better example for this repo might just be a simple options picker. If you agree, I can cook something up soon.

**Comment by mhanberg** [comment]
Okay,  updated the example to be as minimal as I think can be 😅

https://user-images.githubusercontent.com/5523984/180442870-e65cac63-2689-47d2-84d7-c94d87283ff4.mov

.

File: ratatouille.exs [hunk 1]
```diff
@@ -0,0 +1,133 @@
+Mix.install([:ratatouille])
+
+# Selector
+#
+# Basic option chooser using Elixir and Ratatouille
+# 
+# Usage:
+# 
+# $ echo "Option 1\nOption 2\nOption 3" | elixir ratatouille.exs
```
Comment (wojtekmach) [inline ratatouille.exs]:
```suggestion
defmodule Main do
  def main do
    # there isn't a way to return output from the tui, so we use some global state
    Agent.start_link(fn -> nil end, name: :capture_proc)

    Ratatouille.run(RatatouilleSelector,
      quit_events: [
        {:key, Ratatouille.Constants.key(:ctrl_c)},
        {:key, Ratatouille.Constants.key(:esc)}
      ]
    )

    selection = Agent.get(:capture_proc, & &1)

    IO.puts(selection)
  end
end
```

File: ratatouille.exs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+  @up [key(:ctrl_p), key(:arrow_up)]
+  @down [key(:ctrl_n), key(:arrow_down)]
+  @enter key(:enter)
+
+  def init(%{window: %{height: height}}) do
+    lines =
+      IO.read(:eof)
+      |> String.trim()
+      |> String.split("\n")
```
Comment (wojtekmach) [inline ratatouille.exs]:
```suggestion
    lines = ~w(foo bar baz)
```

File: ratatouille.exs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+Ratatouille.run(RatatouilleSelector,
+  quit_events: [
+    {:key, Ratatouille.Constants.key(:ctrl_c)},
+    {:key, Ratatouille.Constants.key(:esc)}
+  ]
+)
+
+selection = Agent.get(:capture_proc, & &1)
+
+IO.puts(selection)
```
Comment (wojtekmach) [inline ratatouille.exs]:
```suggestion
Main.main()
```

**Comment by wojtekmach** [comment]
I've made minor changes. Thank you, this is great!

**Pull request merged**